src/main/java/com/spotify/docker/client/messages/AuthConfig.java

```diff
@@ -109,6 +127,125 @@ public Builder toBuilder() {
+   * This method is package-local so we can test it.
+   * @return a {@link Builder}
+   * @throws IOException
+   */
+  @VisibleForTesting
+  static Builder fromDockerConfig(final Path configPath) throws IOException {
+    return parseDockerConfig(configPath, null);
+  }
+
+  /**
+   * Returns the specified credential from the specified path to the docker file.
+   * This method is package-local so we can test it.
+   * @return a {@link Builder}
+   * @throws IOException
+   */
+  @VisibleForTesting
+  static Builder fromDockerConfig(final Path configPath, final String serverAddress)
+      throws IOException {
+    return parseDockerConfig(configPath, serverAddress);
+  }
+
+  private static Path defaultConfigPath() {
+    final String home = System.getProperty("user.home");
+    final Path dockerConfig = Paths.get(home, ".docker", "config.json");
+    final Path dockerCfg = Paths.get(home, ".dockercfg");
+
+    if (Files.exists(dockerConfig)) {
+      log.debug("Using configfile: {}", dockerConfig);
+      return dockerConfig;
+    } else if (Files.exists(dockerCfg)) {
+      log.debug("Using configfile: {} ", dockerCfg);
+      return dockerCfg;
+    } else {
+      throw new RuntimeException(
+          "Could not find a docker config. Please run 'docker login' to create one");
+    }
+  }
+
+  private static AuthConfig.Builder parseDockerConfig(final Path configPath, String serverAddress)
+      throws IOException {
+    checkNotNull(configPath);
+    final AuthConfig.Builder authBuilder = AuthConfig.builder();
+    final JsonNode authJson = extractAuthJson(configPath);
+
+    if (isNullOrEmpty(serverAddress)) {
+      final Iterator<String> servers = authJson.fieldNames();
+      if (servers.hasNext()) {
+        serverAddress = servers.next();
+      }
+    } else {
+      if (!authJson.has(serverAddress)) {
+        log.error("Could not find auth config for {}. Returning empty builder", serverAddress);
+        return AuthConfig.builder().serverAddress(serverAddress);
+      }
+    }
+
+    final JsonNode serverAuth = authJson.get(serverAddress);
+    if (serverAuth.has("auth")) {
+      authBuilder.serverAddress(serverAddress);
+      final String authString = serverAuth.get("auth").asText();
+      final String[] authParams = Base64.decodeAsString(authString).split(":");
+
+      if (authParams.length == 2) {
+        authBuilder.username(authParams[0].trim());
+        authBuilder.password(authParams[1].trim());
+      } else {
+        log.warn("Failed to parse auth string for {}", serverAddress);
+        return authBuilder;
+      }
+    } else {
+      log.warn("Could not find auth field for {}", serverAddress);
+      return authBuilder;
+    }
+
+    if (serverAuth.has("email")) {
+      authBuilder.email(serverAuth.get("email").asText());
+    }
+
+    return authBuilder;
+  }
+
+  private static JsonNode extractAuthJson(final Path configPath) throws IOException {
+    final ObjectMapper mapper = new ObjectMapper();
```

Comment:
this can be stored as a field; an ObjectMapper instance is threadsafe once constructed